### PR TITLE
Reorganize configuration

### PR DESCRIPTION
**Pull request by pspanja**
This reorganized semantic configuration by renaming and moving it under eZ Platform siteaccess aware configuration.
Pull requests to `site-bundle` and `media-site` will follow.

From the upgrade docs:

Semantic configuration located under eZ Platform siteaccess aware configuration has been renamed and
consolidated under `ng_site_api` key.

- `ng_fallback_to_secondary_content_view` has been renamed to `fallback_to_secondary_content_view`
- `ng_fallback_without_subrequest` has been renamed to `fallback_without_subrequest`
- `ng_richtext_embed_without_subrequest` has been renamed to `richtext_embed_without_subrequest`
- `ng_xmltext_embed_without_subrequest` has been renamed to `xmltext_embed_without_subrequest`
- `ng_named_query` has been renamed to `named_queries`

Semantic configuration located under `netgen_ez_platform_site_api` has been renamed and moved
under `ng_site_api` key under eZ Platform siteaccess aware configuration:

- `override_url_alias_view_action` has been renamed to `site_api_is_primary_content_view`
- `fail_on_missing_fields` has been renamed to `fail_on_missing_field`

Previous configuration:

```yaml
ezpublish:
    system:
        frontend_group:
            ng_fallback_to_secondary_content_view: true
            ng_fallback_without_subrequest: true
            ng_richtext_embed_without_subrequest: true
            ng_xmltext_embed_without_subrequest: true
            ng_named_query: []
netgen_ez_platform_site_api:
    system:
        frontend_group:
            override_url_alias_view_action: true
            use_always_available_fallback: true
            fail_on_missing_fields: true
            render_missing_field_info: true
            named_objects: []
```

New configuration:

```yaml
ezpublish:
    system:
        frontend_group:
            ng_site_api:
                site_api_is_primary_content_view: true
                fallback_to_secondary_content_view: true
                fallback_without_subrequest: true
                richtext_embed_without_subrequest: true
                xmltext_embed_without_subrequest: true
                use_always_available_fallback: true
                fail_on_missing_field: true
                render_missing_field_info: true
                named_objects: []
                named_queries: []
```

Key for Site API content view configuration has been renamed from `ngcontent_view` to
`ng_content_view`:

Previous configuration:

```yaml
ezpublish:
    system:
        frontend_group:
            ngcontent_view: []
```

New configuration:

```yaml
ezpublish:
    system:
        frontend_group:
            ng_content_view: []
```

Named object configuration has renamed keys used for named object types:

- `content` has been renamed to `content_items`
- `location` has been renamed to `locations`
- `tag` has been renamed to `tags`

Previous configuration:

```yaml
netgen_ez_platform_site_api:
    system:
        frontend_group:
            named_objects:
                content:
                    certificate: 123
                location:
                    home: 2
                tag:
                    colors: 456
```

New configuration:

```yaml
ezpublish:
    system:
        frontend_group:
            ng_site_api:
                named_objects:
                    content:
                        certificate: 123
                    locations:
                        home: 2
                    tags:
                        colors: 456
```

### CODE REVIEW

**Comment by emodric** [comment]
I like the idea of renaming the config and putting it under `ezpublish` namespace to be as close as possible to eZ view config (to which most of this config is related to, anyway).

I do have two suggestions: put everything (except view config) under a special `ng_site_api` key, to avoid repeating `ng_` prefixes.

Name the view config `ng_content_view` (singular form), and leave it outside of `ng_site_api` key to be as close as possible to default eZ view configuration (thus reducing the complexity of rewriting existing eZ view rules to Site API ones).

The config would look like:

```yaml
frontend_group:
    ng_site_api:
        set_site_api_as_primary_content_view: true
        fallback_to_secondary_content_view: true
        fallback_without_subrequest: true
        fail_on_missing_field: false
        render_missing_field_info: true
        named_objects:
            locations:
                home: '@=config("content.tree_root.location_id")'
                site_info: '@=config("content.site_info.location_id")'
        named_queries: []
    ng_content_view:
        full:
            ng_feedback_form:
                template: "@ezdesign/content/full/ng_feedback_form.html.twig"
                controller: "netgen_information_collection.controller:displayAndHandle"
                match:
                    Identifier\ContentType: ng_feedback_form
            common:
                template: "@ezdesign/content/full/{content_type}.html.twig"
                match:
                    Identifier\ContentType:
                        - ng_article
                        - ng_audio
                        - ng_blog_post
                        - ng_frontpage
                        - ng_gallery
                        - ng_htmlbox
                        - ng_news
                        - ng_recipe
                        - ng_test
                        - ng_topic
                        - ng_video
            catch_all:
                temporary_redirect: "@=namedLocation('home')"
                match: ~
```

**Comment by hknezevic** [comment]
@pspanja 
Proposal by @emodric makes more sense to me. I like the idea of having everything under ezpublish block, and I also prefer to have a separate siteaccess-aware config block for all related configurations.

What is the reasoning for changing ngcontent_view to ng_content_views ? Personally I prefer for it to be as close to default eZ naming as possible

**Comment by pspanja** [comment]
@hknezevic the reason is to make it in line with the rest of the config (named objects and queries). But I agree with Edi's suggestion, will update as suggested.

**Comment by pspanja** [comment]
Updated as suggested in https://github.com/netgen/ezplatform-site-api/pull/169/commits/1b5b66e59f08bd4052118b0e7f6a9039989f5252
PRs on media-site and site-bundle will follow.

**Comment by pspanja** [comment]
Also removed XmlText support in https://github.com/netgen/ezplatform-site-api/pull/169/commits/74f3873c3b7f76d83c876e130f22ae9df119389a

File: docs/upgrades/upgrade_350_400.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+.. code-block:: yaml
+
+    ezpublish:
+        system:
+            frontend_group:
+                ng_content_view: []
+
+Named object configuration has renamed keys used for named object types:
+
+- ``content`` has been renamed to ``content_items``
```
Comment (emodric) [inline docs/upgrades/upgrade_350_400.rst]: How about we leave this named `content`? I'm worried that `content_items` name might clash with "block items" from Layouts. Besides, word `content` itself can describe multiple units of content objects.

Comment (pspanja) [reply docs/upgrades/upgrade_350_400.rst]: I wanted to be in line with the official language here, but content is also fine for me. Will update.

Comment (pspanja) [reply docs/upgrades/upgrade_350_400.rst]: Updated in https://github.com/netgen/ezplatform-site-api/pull/169/commits/d8e42c036d5cef7b3339421dc7b52197a70f575f

**Review by emodric — APPROVED** [review]
Looks good to me :+1:

**Comment by pspanja** [comment]
Thanks @emodric

**Pull request merged**